eng/common/TestResources/deploy-test-resources.yml

```diff
@@ -27,13 +42,31 @@ steps:
         ${{ parameters.SubscriptionConfiguration }}
       "@ | ConvertFrom-Json -AsHashtable;
 
+      if ($subscriptionConfiguration.ArmTemplateParameters) {
```

Comment:
While you're ad it, I would appreciate if you didn't restrict specifying additional parameters to only CIs. My intent was always for tests.yml to be able to provide them as well. Since you're merging hashtables, now is a great opportunity to provide for that.